editor/import/editor_importer_bake_reset.cpp

```diff
@@ -0,0 +1,201 @@
+		}
+		queue.pop_front();
+	}
+}
+
+void BakeReset::_align_animations(AnimationPlayer *p_ap, const Map<StringName, BakeResetRestBone> &r_rest_bones) {
+	ERR_FAIL_NULL(p_ap);
+	List<StringName> anim_names;
+	p_ap->get_animation_list(&anim_names);
+	for (List<StringName>::Element *anim_i = anim_names.front(); anim_i; anim_i = anim_i->next()) {
+		Ref<Animation> a = p_ap->get_animation(anim_i->get());
+		ERR_CONTINUE(a.is_null());
+		for (Map<StringName, BakeResetRestBone>::Element *rest_bone_i = r_rest_bones.front(); rest_bone_i; rest_bone_i = rest_bone_i->next()) {
+			int track = a->find_track(NodePath(rest_bone_i->key()));
+			if (track == -1) {
+				continue;
+			}
+			int new_track = a->add_track(Animation::TYPE_TRANSFORM3D);
+			NodePath new_path = NodePath(rest_bone_i->key());
+			BakeResetRestBone rest_bone = rest_bone_i->get();
+			a->track_set_path(new_track, new_path);
+			for (int key_i = 0; key_i < a->track_get_key_count(track); key_i++) {
+				Vector3 loc;
+				Quaternion rot;
+				Vector3 scale;
+				Error err = a->transform_track_get_key(track, key_i, &loc, &rot, &scale);
+				ERR_CONTINUE(err);
+				real_t time = a->track_get_key_time(track, key_i);
+				rot.normalize();
+				loc = loc - rest_bone.loc;
+				rot = rest_bone.rest_delta.get_rotation_quaternion().inverse() * rot;
+				rot.normalize();
+				scale = Vector3(1, 1, 1) - (rest_bone.rest_delta.get_scale() - scale);
+				a->transform_track_insert_key(new_track, time, loc, rot, scale);
+			}
+			a->remove_track(track);
+		}
+	}
+}
+
+void BakeReset::_fetch_reset_animation(AnimationPlayer *p_ap, Map<StringName, BakeResetRestBone> &r_rest_bones, const String &p_bake_anim) {
+	ERR_FAIL_NULL(p_ap);
+	List<StringName> anim_names;
+	p_ap->get_animation_list(&anim_names);
+	Node *root = p_ap->get_owner();
+	ERR_FAIL_NULL(root);
+	Ref<Animation> a = p_ap->get_animation(p_bake_anim);
+	ERR_FAIL_NULL(a);
+	for (int32_t track = 0; track < a->get_track_count(); track++) {
+		NodePath path = a->track_get_path(track);
+		String string_path = path;
+		Skeleton3D *skeleton = root->cast_to<Skeleton3D>(root->get_node(string_path.get_slice(":", 0)));
+		if (!skeleton) {
+			continue;
+		}
+		String bone_name = string_path.get_slice(":", 1);
+		for (int key_i = 0; key_i < a->track_get_key_count(track); key_i++) {
+			Vector3 loc;
+			Quaternion rot;
+			Vector3 scale;
+			Error err = a->transform_track_get_key(track, key_i, &loc, &rot, &scale);
+			ERR_CONTINUE(err);
+			rot.normalize();
+			Basis rot_basis = Basis(rot, scale);
+			BakeResetRestBone rest_bone;
+			rest_bone.rest_delta = rot_basis;
+			rest_bone.loc = loc;
+			r_rest_bones[StringName(String(skeleton->get_owner()->get_path_to(skeleton)) + ":" + bone_name)] = rest_bone;
+			break;
+		}
+	}
+}
+
+void BakeReset::_fix_skeleton(Skeleton3D *p_skeleton, Map<StringName, BakeReset::BakeResetRestBone> &r_rest_bones) {
+	int bone_count = p_skeleton->get_bone_count();
+
+	//First iterate through all the bones and create a RestBone
```

Comment:
Please use this style for comments (space after `//`, capital letter, final dot):
```suggestion
	// First iterate through all the bones and create a RestBone.
```